src/promptflow/tests/sdk_cli_test/e2etests/test_flow_as_func.py

```diff
@@ -137,3 +141,101 @@ def test_non_secret_connection(self):
         # we only raise error when there are scrubbed secrets in connection
         f.context.connections = {"hello_node": {"connection": CustomConnection(secrets={})}}
         f(text="hello")
+
+    def test_flow_context_cache(self):
+        # same flow context has same hash
+        assert hash(FlowContext()) == hash(FlowContext())
+        # getting executor for same flow will hit cache
+        flow1 = load_flow(f"{FLOWS_DIR}/print_env_var")
```

Comment:
For a long running scenario, I'm not sure if print_env_var cache will be automatically picked up if I changed the dag/configuration etc.

If I change it, but the cache will never be hit, please consider expiration of cache values.
The cache should not

Reply:
See `test_flow_context_cache_not_hit`, the cache won't hit if flow/context changed.